watson/utils.py

```diff
@@ -41,25 +41,9 @@ def format_timedelta(delta):
     """
     Return a string roughly representing a timedelta.
     """
-    seconds = int(delta.total_seconds())
-    neg = seconds < 0
-    seconds = abs(seconds)
-    total = seconds
-    stems = []
-
-    if total >= 3600:
-        hours = seconds // 3600
-        stems.append('{}h'.format(hours))
-        seconds -= hours * 3600
-
-    if total >= 60:
-        mins = seconds // 60
-        stems.append('{:02}m'.format(mins))
-        seconds -= mins * 60
-
-    stems.append('{:02}s'.format(seconds))
-
-    return ('-' if neg else '') + ' '.join(stems)
+    neg = '-' if int(delta.total_seconds()) < 0 else ''
+    res = "{}h {}m {}s".format(*str(abs(delta)).split(":"))
```

Comment:
The python datetime module actually takes care of the formatting for us.
See the str() function docs:
https://docs.python.org/3/library/datetime.html#timedelta-objects.

Thanks!
Scott